kernel/src/sched.rs

```diff
@@ -900,4 +732,236 @@ impl Kernel {
+                        driver_number,
+                        subdriver_number,
+                        arg0,
+                        arg1,
+                        res,
+                    );
+                }
+                process.set_syscall_return_value(res);
+            }
+            Syscall::ReadWriteAllow {
+                driver_number,
+                subdriver_number,
+                allow_address,
+                allow_size,
+            } => {
+                let res = platform.with_driver(driver_number, |driver| match driver {
+                    Some(d) => process.allow_readwrite(allow_address, allow_size, &|appslice| {
+                        d.allow_readwrite(process.appid(), subdriver_number, appslice)
+                    }),
+                    None => GenericSyscallReturnValue::AllowReadWriteFailure(
+                        ErrorCode::NOSUPPORT,
+                        allow_address,
+                        allow_size,
+                    ),
+                });
+
+                if config::CONFIG.trace_syscalls {
+                    debug!(
+                        "[{:?}] read-write allow({:#x}, {}, @{:#x}, {:#x}) = {:?}",
+                        process.appid(),
+                        driver_number,
+                        subdriver_number,
+                        allow_address as usize,
+                        allow_size,
+                        res
+                    );
+                }
+                process.set_syscall_return_value(res);
+            }
+            Syscall::ReadOnlyAllow {
+                driver_number,
+                subdriver_number,
+                allow_address,
+                allow_size,
+            } => {
+                let res = platform.with_driver(driver_number, |driver| match driver {
+                    Some(d) => process.allow_readonly(allow_address, allow_size, &|appslice| {
+                        d.allow_readonly(process.appid(), subdriver_number, appslice)
+                    }),
+                    None => GenericSyscallReturnValue::AllowReadOnlyFailure(
+                        ErrorCode::NOSUPPORT,
+                        allow_address,
+                        allow_size,
+                    ),
+                });
+
+                if config::CONFIG.trace_syscalls {
+                    debug!(
+                        "[{:?}] read-only allow({:#x}, {}, @{:#x}, {:#x}) = {:?}",
+                        process.appid(),
+                        driver_number,
+                        subdriver_number,
+                        allow_address as usize,
+                        allow_size,
+                        res
+                    );
+                }
+
+                process.set_syscall_return_value(res);
+            }
+            Syscall::Exit {
+                which,
+                completion_code,
+            } => match which {
+                0 => process.terminate(completion_code as u32),
```

Comment:
```suggestion
                // The process called exit, and the process indicates what it
                // wants the kernel to do next. If `which` is 0, then the process is
                // finished and wants to exit entirely. If `which` is 1, then the process
                // can't continue for some reason, but would like the kernel to try to
                // restart it.
                //
                // `process.rs` decides how the kernel actually wants to handle those
                // conventions, since the kernel is not bound to do what the process
                // asked.
                0 => process.terminate(completion_code as u32),
```

Reply:
This just repeats what is said in the syscall TRD. Better to just say which variant it is and refer to the TRD if necessary. Repeated documentation can be inconsistent.